aten/src/ATen/native/cuda/SummaryOps.cu

```diff
@@ -0,0 +1,57 @@
+#include "ATen/ATen.h"
+#include "ATen/cuda/CUDAApplyUtils.cuh"
+
+#include <type_traits>
+
+namespace at { namespace native {
+
+///////////////// bincount /////////////////
+namespace {
+template <typename weights_t, typename integral_t>
+Tensor _bincount_cuda_template(
+    const Tensor& self,
+    const Tensor& weights,
+    int64_t minlength) {
+  if (minlength < 0) {
+    AT_ERROR("minlength should be >= 0");
+  }
+  if (self.dim() != 1 || self.numel() == 0 ||
+      !isIntegralType(self.type().scalarType()) ||
+      (!std::is_same<integral_t, uint8_t>::value &&
+       *self.min().toBackend(kCPU).data<integral_t>() < 0)) {
```

Comment:
`min()` will launch another kernel... I think it's okay if we forego this safety check and add it as an assert inside a kernel